sktime/annotation/hidalgo.py

```diff
@@ -0,0 +1,699 @@
+
+                    if g > gmax:
+                        gmax = g
+                    gg[k1] = g
+
+                gg = [np.exp(gg[k1] - gmax) for k1 in range(K)]
+
+                prob = p * d * mu[i] ** (-(d + 1)) * gg
+                prob /= prob.sum()
+
+                while stop is False:
+                    r1 = int(np.floor(self._rng.random() * K))  # random sample for Z
+                    r2 = self._rng.random()  # random number for accepting
+
+                    if prob[r1] > r2:
+                        stop = True
+                        # minus values
+                        NN[Z[i]] -= 1
+                        a1[Z[i]] -= 1
+                        c1[Z[i]] -= 1
+                        V[Z[i]] -= np.log(mu[i])
+                        b1[Z[i]] -= np.log(mu[i])
+                        # change Z, add values
+                        Z[i] = r1
+                        NN[Z[i]] += 1
+                        a1[Z[i]] += 1
+                        c1[Z[i]] += 1
+                        V[Z[i]] += np.log(mu[i])
+                        b1[Z[i]] += np.log(mu[i])
+
+            return Z, NN, a1, c1, V, b1
+
+        def sample_likelihood(p, d, Z, N_in, zeta, NN):
+
+            N = self.N
+            mu = self.mu
+
+            lik0 = 0
+            for i in range(N):
+                lik0 = (
+                    lik0
+                    + np.log(p[Z[i]])
+                    + np.log(d[Z[i]])
+                    - (d[Z[i]] + 1) * np.log(mu[i])
+                )
+
+            lik1 = lik0 + np.log(zeta / (1 - zeta)) * N_in
+
+            for k1 in range(K):
+                lik1 = lik1 - (NN[k1] * np.log(Zpart(N, NN[k1], zeta, q)))
+
+            return lik0, lik1
+
+        for it in range(Niter):
+
+            d = sample_d(K, a1, b1)
+            sampling = np.append(sampling, d)
+
+            (p, pp) = sample_p(K, p, pp, c1)
+            sampling = np.append(sampling, p[: K - 1])
+            sampling = np.append(sampling, (1 - pp))
+
+            zeta = sample_zeta(K, zeta, use_Potts, estimate_zeta, q, NN, f1, it)
+            sampling = np.append(sampling, zeta)
+
+            Z, NN, a1, c1, V, b1 = sampling_Z(Z, NN, a1, c1, V, b1, zeta, fixed_Z, q)
+            sampling = np.append(sampling, Z)
+
+            N_in, f1 = self.update_zeta_prior(Z)
+
+            lik = sample_likelihood(p, d, Z, N_in, zeta, NN)
+            sampling = np.append(sampling, lik)
+
+        return sampling
+
+    def _fit(self, X, y=None):
+        """There is no need to fit a model for HidAlgo.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Training data to fit model to (time series).
+        y : pd.Series, optional
+            Ground truth annotations for training, not specified as HidAlgo
+            is unsupervised.
+
+        Returns
+        -------
+        self
+        """
+        seed = self.seed
+        self._rng = check_random_state(seed)
+        return self
```

Comment:
lets remove `_fit` then and add the tag `"fit_is_empty": True` and add a line `self._rng = check_random_state(self.seed)` there

Reply:
wait, put `self._rng = check_random_state(self.seed)` where?

Reply:
into `predict()`, or `transform()` if we go for a transformer